switools/signaturelib.py

```diff
@@ -12,4 +13,23 @@ def getSigFileName( swiFile ):
       return SWIX_SIG_FILE_NAME
    return SWI_SIG_FILE_NAME
 
+def getOptimizations( swi, workDir ):
+   ret = os.system( "set -e; swi=$(readlink -f %s); cd %s; "
+                    "unzip -q -o $swi swimSqshMap" % ( swi, workDir ) )
+   if ret:
+      return None # legacy image
+   optims = []
+   with open( "%s/swimSqshMap" % workDir ) as f:
+      for line in f:
+         optim, _ = line.split( "=", 1 )
+         optims.append( optim )
+   os.system( "rm %s/swimSqshMap" % workDir )
+   return optims
 
+def extractSwadapt( swi, workDir ):
+   ret = os.system( "set -e; image=$(readlink -f %s); cd %s;"
+                    "unzip -o -q $image swadapt" % ( swi, workDir ) )
+   if ret:
+      print( "Error: '%s' does not contain the 'swadapt' utility" % swi )
+      shutil.rmtree( workDir )
```

Comment:
I think it's safer to remove the workDir from outside this function? And also it looks like it's unnecessary currently since this function is always called in the try/finally with shutil.rmtree in the finally.

Reply:
Well, at that point, the only thing that needs cleanup is that temp dir, so exiting right there is ok, and makes for simpler code (no need to create an exception that is later caught and pretty printed). Maybe just add a comment on the call that it might never return?

Reply:
Also this is not an operator error, and has more of an assert kind of quality (unrecoverable): a multi-fs image that does not contain swadapt should not exist in the wild, so I would not find it warranted to jump over backwards with exceptions.

Reply:
using tempfile.TemporaryDirectory in the upper level now.